From 52d9a41642368d802394bed896f48778e60cb1c5 Mon Sep 17 00:00:00 2001
From: Chen Liangjun <b36089@freescale.com>
Date: Wed, 1 Aug 2012 13:57:27 +0800
Subject: ENGR00219016-2 MX6Q ARD: Set HDMI event as event 2 of SDMA event

If MX6q chip version is bigger than TO1.2, Select HDMI SDMA
request as SDMA event 2 for MX6Q ARM2 board. SDMA event 2 can be
configured HDMI or IPU.

Signed-off-by: Chen Liangjun <b36089@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_sabreauto.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-mx6/board-mx6q_sabreauto.c b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
index b80a4bc..ff031e4 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabreauto.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
@@ -910,6 +910,10 @@ static void hdmi_init(int ipu_id, int disp_id)
 
 	/* GPR3, bits 2-3 = HDMI_MUX_CTL */
 	mxc_iomux_set_gpr_register(3, 2, 2, hdmi_mux_setting);
+
+	/* Set HDMI event as SDMA event2 while Chip version later than TO1.2 */
+	if ((mx6q_revision() > IMX_CHIP_REVISION_1_1))
+		mxc_iomux_set_gpr_register(0, 0, 1, 1);
 }
 
 /* On mx6x sabreauto board i2c2 iomux with hdmi ddc,
-- 
1.7.9.5

